packages/core/src/transport/WsOutboundTransporter.ts

```diff
@@ -1,42 +1,45 @@
 import type { Agent } from '../agent/Agent'
 import type { Logger } from '../logger'
+import type { ConnectionRecord } from '../modules/connections'
 import type { OutboundPackage } from '../types'
 import type { OutboundTransporter } from './OutboundTransporter'
+import type { OutboundWebSocketClosedEvent } from './TransportEventTypes'
 import type WebSocket from 'ws'
 
 import { AgentConfig } from '../agent/AgentConfig'
+import { EventEmitter } from '../agent/EventEmitter'
 import { AriesFrameworkError } from '../error/AriesFrameworkError'
 
+import { TransportEventTypes } from './TransportEventTypes'
+
 export class WsOutboundTransporter implements OutboundTransporter {
   private transportTable: Map<string, WebSocket> = new Map<string, WebSocket>()
   private agent!: Agent
   private logger!: Logger
+  private eventEmitter!: EventEmitter
   private WebSocketClass!: typeof WebSocket
-  private continue!: boolean
-  private recursiveBackOff: { [socketId: string]: number } = {}
   public supportedSchemes = ['ws', 'wss']
 
   public async start(agent: Agent): Promise<void> {
     this.agent = agent
-    this.continue = true
     const agentConfig = agent.injectionContainer.resolve(AgentConfig)
 
     this.logger = agentConfig.logger
+    this.eventEmitter = agent.injectionContainer.resolve(EventEmitter)
     this.logger.debug('Starting WS outbound transport')
     this.WebSocketClass = agentConfig.agentDependencies.WebSocketClass
   }
 
   public async stop() {
     this.logger.debug('Stopping WS outbound transport')
-    this.continue = false
     this.transportTable.forEach((socket) => {
       socket.removeEventListener('message', this.handleMessageEvent)
       socket.close()
     })
   }
 
   public async sendMessage(outboundPackage: OutboundPackage) {
-    const { payload, endpoint } = outboundPackage
+    const { payload, endpoint, connection } = outboundPackage
```

Comment:
I personally would really like to see this be connectionId as a string vs sending the whole connectionRecord, since that starts to really tie our transports into the inner workings of the Framework more than they are now.

Reply:
That's an interesting point. But `connectionRecrod` is already part of the public API of the framework 🤔 If you think we're exposing too much framework internal info via `connectionRecord`, maybe, we should introduce some domain object hiding unnecessary details.